### PR TITLE
Add dropout option for BERT Pooler

### PR DESCRIPTION
**Pull request by luckytoilet**
This adds an optional dropout to BERT after the pooling operation but before the final feedforward. This mirrors `BertForClassification`, which has a dropout at this step.

### CODE REVIEW

File: allennlp/modules/seq2vec_encoders/bert_pooler.py [hunk 1]
```diff
@@ -26,18 +27,25 @@ class BertPooler(Seq2VecEncoder):
...
+    dropout : ``float``, optional, (default = 0.0)
+        Amount of dropout to apply after pooling
     requires_grad : ``bool``, optional, (default = True)
         If True, the weights of the pooler will be updated during training.
         Otherwise they will not.
     """
-    def __init__(self, pretrained_model: Union[str, BertModel], requires_grad: bool = True) -> None:
+    def __init__(self,
+                 pretrained_model: Union[str, BertModel],
+                 dropout: float = 0.0,
```
Comment (DeNeutoy) [inline allennlp/modules/seq2vec_encoders/bert_pooler.py]: Please could you change the position of this argument to be last, so that it doesn't introduce a breaking change?

Comment (luckytoilet) [reply allennlp/modules/seq2vec_encoders/bert_pooler.py]: Ok, done
